File: ruby/QBO_request.rb
require 'oauth/request_proxy/typhoeus_request'
require 'json'
require 'yaml'

CONFIG = YAML.load_file("config.yml")

def api_call(access_token, realm_id)
	url =  CONFIG['accounting_base_url'] + 'v3/company/' + realm_id + '/companyinfo/' + realm_id+'?minorversion=12'
	options = {
		method: :get,
		headers: { 
			'Content-Type' => 'application/json',
			'Accept' => 'application/json',
			'Authorization' => 'Bearer ' + access_token
		}
	}
	req = Typhoeus::Request.new(url, options)
	req.run
	response = req.response
	puts 'Test API call'
	puts response.code 
	puts response.headers 
	puts response.body 
	JSON.parse(response.body)
end

File: python/QBO_request.py
import requests
import json

with open('config.json', 'r') as f:
        config = json.load(f)

def api_call(access_token, realmId):
    '''Test a GET API request with OAuth2 access token'''
    url = config['Accounting_base_url'] + 'v3/company/' + realmId + '/companyinfo/' + realmId+'?minorversion=12'
    headers = { 
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': 'Bearer '+ access_token 
    }
    r = requests.get(url, headers=headers)
    print ('Test API call:')
    print ('Status code: '+str(r.status_code))
    #print (r.headers)
    print (r.json())
    return r.json()